src/zenml/models/v2/core/artifact.py

```diff
@@ -225,160 +115,24 @@ def tags(self) -> List[TagResponseModel]:
         """
         return self.get_metadata().tags
 
-    @property
-    def visualizations(
-        self,
-    ) -> Optional[List["ArtifactVisualizationResponse"]]:
-        """The `visualizations` property.
-
-        Returns:
-            the value of the property.
-        """
-        return self.get_metadata().visualizations
-
-    @property
-    def run_metadata(self) -> Dict[str, "RunMetadataResponse"]:
-        """The `metadata` property.
-
-        Returns:
-            the value of the property.
-        """
-        return self.get_metadata().run_metadata
-
-    @property
-    def materializer(self) -> Source:
-        """The `materializer` property.
-
-        Returns:
-            the value of the property.
-        """
-        return self.get_metadata().materializer
-
-    @property
-    def data_type(self) -> Source:
-        """The `data_type` property.
-
-        Returns:
-            the value of the property.
-        """
-        return self.get_metadata().data_type
-
-    @property
-    def has_custom_name(self) -> bool:
-        """The `has_custom_name` property.
-
-        Returns:
-            the value of the property.
-        """
-        return self.get_metadata().has_custom_name
-
     # Helper methods
     @property
-    def step(self) -> "StepRunResponse":
-        """Get the step that produced this artifact.
-
-        Returns:
-            The step that produced this artifact.
-        """
-        from zenml.artifacts.utils import get_producer_step_of_artifact
-
-        return get_producer_step_of_artifact(self)
-
-    @property
-    def run(self) -> "PipelineRunResponse":
-        """Get the pipeline run that produced this artifact.
+    def versions(self) -> Dict[str, "ArtifactVersionResponse"]:
```

Comment:
I was wondering if we should add the versions to the metadata of the `ArtifactResponse`? Is there a specific reason why you didn't?

Reply:
Be careful with this one though. You can only put it in the `metadata` of the response. `ArtifactVersionResponse`s already have an `ArtifactResponse` in their `body`. If you put a list `ArtifactVersionResponse`s in the body of the `ArtifactResponse`s, it will create a loop. That's why it can only go into the `metadata`.